src/pymmcore_widgets/_mda/_channel_table_widget.py

```diff
@@ -109,22 +122,53 @@ def __init__(
 
         group_layout.addWidget(wdg, 0, 1)
 
-    def setChannelGroup(self, group: str) -> None:
-        """Set current channel group."""
-        if group:
-            avail = self._mmc.getAvailableConfigGroups()
-            if group not in avail:
-                raise ValueError(
-                    f"ChannelGroup '{group}' not available config groups: {avail!r}."
-                )
-        self._channel_group = group
+        self._mmc.events.systemConfigurationLoaded.connect(self.clear)
+        self._mmc.events.configGroupDeleted.connect(self._on_group_deleted)
+        self._mmc.events.configDeleted.connect(self._on_config_deleted)
+
+        self.destroyed.connect(self._disconnect)
+
+    def _on_group_deleted(self, group: str) -> None:
+        """Remove rows that are using channels from the deleted group."""
+        row = 0
+        for ch in self.value():
+            if ch["group"] == group:
+                self._table.removeRow(row)
+            else:
+                row += 1
+
+    def _on_config_deleted(self, group: str, config: str) -> None:
+        """Remove deleted config from channel combo if present."""
+        for row in range(self._table.rowCount()):
+            combo = cast(QComboBox, self._table.cellWidget(row, 0))
+            items = [combo.itemText(ch) for ch in range(combo.count())]
+            if group == combo.whatsThis() and config in items:
+                combo.clear()
+                combo.addItems(self._mmc.getAvailableConfigs(group))
+                if self._mmc.getChannelGroup() != config:
+                    combo.setCurrentText(self._mmc.getChannelGroup())
 
     def channelGroup(self) -> str:
         """Return current channel group."""
         return self._channel_group
```

Comment:
does this attribute even exist anymore?

Reply:
> let's say someone had a configuration that had _two_ config groups that were meaningful as channel groups... and they wanted to have a channel table widget dedicated to each of them. Wouldn't this prevent that, since now the channel table widget is _obligated_ to follow the current group?

Yes, that's true! What about if we keep both versions? one that is more general (the current one, which we can call something like `GeneralChannelTable`) and this one (`ChannelTable`) which is more similar to micromanager...or we can implement it only in `napari-micromanager` if you think it is better!

> also, i think there are some latent bugs here, since `_channel_group` is accessed in places but isn't actually set anywhere.

That's true, I forgot to remove the variable in few places...Also at the moment the micromanager `ChannelGroup` property is not controlled by the `ChannelTable` widget itself but is only handled by the `ChannelGroupWidget` that I have added in this PR (`_channel_group_widget.py`).

About testing this widget, I added some tests to check that everything worked as I wanted - `test_channel_table_widget.py`.